src/main/java/core/basesyntax/dao/animal/AnimalDaoImpl.java

```diff
@@ -3,20 +3,51 @@
 import core.basesyntax.dao.AbstractDao;
 import core.basesyntax.model.zoo.Animal;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
 
 public class AnimalDaoImpl extends AbstractDao implements AnimalDao {
+
     public AnimalDaoImpl(SessionFactory sessionFactory) {
+
         super(sessionFactory);
     }
 
     @Override
     public Animal save(Animal animal) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(animal);
+            transaction.commit();
+            return animal;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Animal has not been added/n", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Animal> findByNameFirstLetter(Character character) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            Query<Animal> query = session.createQuery(
+                    "FROM Animal a "
+                            + "WHERE a.name LIKE :name ",
+                    Animal.class);
```

Comment:
```suggestion
            Query<Animal> query = session.createQuery(
                    "FROM Animal a WHERE a.name LIKE :name ",
                    Animal.class);
```
I would suggest smth like this. Choose any of the given options)

Reply:
Done